### PR TITLE
feat(SurfaceEdges): extract edges API from SurfaceMesh

### PR DESCRIPTION
**Pull request by BotellaA**
BREAKING CHANGE: rewrite of all the SurfaceMesh edges API

### CODE REVIEW

File: src/geode/mesh/CMakeLists.txt [hunk 1]
```diff
@@ -66,6 +68,7 @@ add_geode_library(
         "core/register_mesh.cpp"
         "core/regular_grid.cpp"
         "core/solid_mesh.cpp"
+        "core/surface_edges.cpp"
         "core/surface_mesh.cpp"
         "core/tetrahedral_solid.cpp"
```
Comment (panquez) [inline src/geode/mesh/CMakeLists.txt:73]: Is `core/detail/edge_view_impl.h` to be removed?

File: src/geode/mesh/builder/surface_edges_builder.cpp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    {
+        edges_->remove_edge( std::move( edge_vertices ), {} );
+    }
+
+    template < index_t dimension >
+    std::vector< index_t >
+        SurfaceEdgesBuilder< dimension >::remove_isolated_edges()
+    {
+        return edges_->remove_isolated_edges( {} );
+    }
```
Comment (panquez) [inline src/geode/mesh/builder/surface_edges_builder.cpp:90]: to me removed/deleted :p

File: tests/mesh/test-triangulated-surface.cpp [hunk 3]
```diff
@@ -56,8 +58,8 @@ void test_create_polygons( const geode::TriangulatedSurface3D& surface,
     builder.set_polygon_vertex( { 0, 0 }, 0 );
     builder.set_polygon_vertex( { 0, 1 }, 1 );
     builder.set_polygon_vertex( { 0, 2 }, 2 );
-    builder.delete_isolated_edges();
-    OPENGEODE_EXCEPTION( surface.nb_edges() == 7,
+    builder.edges_builder().delete_isolated_edges();
+    OPENGEODE_EXCEPTION( surface.edges().nb_edges() == 7,
         "[Test] TriangulatedSurface should have 7 edges" );
```
Comment (panquez) [inline tests/mesh/test-triangulated-surface.cpp:63]: could you remove this code? enable edges after this function and test number of edges?

Comment (panquez) [reply tests/mesh/test-triangulated-surface.cpp:63]: or better test it after setting polygon adjacencies.

**Review by panquez — APPROVED** [review]
LGTM... only a suggestion for tests.

File: src/geode/mesh/CMakeLists.txt (hunk 1 above)
Comment (BotellaA) [reply src/geode/mesh/CMakeLists.txt:73]: it was'nt added... no comment

File: tests/mesh/test-triangulated-surface.cpp (hunk 3 above)
Comment (BotellaA) [reply tests/mesh/test-triangulated-surface.cpp]: done

**Pull request merged**